cmd/service-controller/definition_monitor.go

```diff
@@ -197,9 +197,17 @@ func updateAnnotatedServiceDefinition(actual *types.ServiceInterface, desired *t
 	if actual.Origin != "annotation" {
 		return false
 	}
-	if actual.Protocol != desired.Protocol || !reflect.DeepEqual(actual.Ports, desired.Ports) {
```

Comment:
This seems to have the same effect to me. The new test also passes without changing it.

What seems to be wrong here for me is the fact that, if any of the if statements in the
beginning of the function returns true, the final section of this function, which actually
updates the targets in the desired argument, is not being executed, therefore if ports
AND targets have changed, the desired targets are not being evaluated.

Reply:
You are right, the new test does not change the order of the ports and the new test logic incorrectly assumes the ports are in a map, rather than a list. Both fixed now.